src/App.js

```diff
@@ -1,39 +1,190 @@
+                item.inWishList=true
+              }
+              return x
+            })
+            return item
+          })
+        product === "placeholder"
+          ? setFilteredArray(filteredProduct1)
+          : setFilteredArray(
+              filteredProduct1.filter(
+                ({ product_type }) => product === product_type
+              )
+            )
+           
 
-    </div>
-  );
-}
+      })
+
+  }, [arrayOfObjectWish, product])
 
-export default App;
+  const getIdObject = (product) => {
+    const exist = arrayOfObjectCart.find((x) => x.id === product.id)
+    if (exist) {
+      const newCartItems = arrayOfObjectCart.map((x) =>
+        x.id === product.id ? { ...exist, qty: exist.qty + 1 } : x
+      )
+      setArrayOfObjectCart(newCartItems)
+    } else {
+      const newCartItems = [...arrayOfObjectCart, { ...product, qty: 1 }]
+      setArrayOfObjectCart(newCartItems)
+    }
+  }
+  const getWishIdObject = (event, product) => {
+    // toggle the boolean value
+   product.inWishList=!product.inWishList
+    filteredArray.filter((x)=> {
+      if(x.id===product.id){
+        x.inWishList=product.inWishList
+      }
+      return x
+    })
+    const exist = arrayOfObjectWish.find((x) => x.id === product.id)
+    // If heart is unclick, remove that item from wishList
+    if (exist && event.target.className === `heartAnimation ${product.id} animate`) {
+      const findSelectedProduct = arrayOfObjectWish.filter((item) => item.id !== product.id)
+      setArrayOfObjectWish(findSelectedProduct)
+      // If heart is clicked, add that item from wishList
+    }else if (!exist && event.target.className === `heartAnimation ${product.id}`) {
+      const newCartItems = [...arrayOfObjectWish, { ...product }]
+      setArrayOfObjectWish(newCartItems)
+    }
+  }
 
+  const removeFromWish = (e, individual) => {
+    individual.inWishList = false
+    const newFilteredArray = filteredArray.filter((x) => {
+      
+      if(x.id=== individual.id){
+        x.inWishList = false
+      }
+      return x
 
+    })
+    const filteredWishArray = arrayOfObjectWish.filter((x) => {
+      return individual.id !== x.id
+    })
+    setFilteredArray(newFilteredArray)
+    setArrayOfObjectWish(filteredWishArray)
+  }
 
-// PSEUDO CODE
+    return (
+      <Routes>
+        <Route
+          path='/'
+          element={
+            <>
+              <div className='App'>
+                <HeaderNav
+                  arrayOfObjectCart={arrayOfObjectCart}
+                  setArrayOfObjectCart={setArrayOfObjectCart}
+                  isCartClicked={isCartClicked}
+                  setIsCartClicked={setIsCartClicked}
+                  isHeartClicked={isHeartClicked}
+                  setIsHeartClicked={setIsHeartClicked}
+                  arrayOfObjectWish={arrayOfObjectWish}
+                  getIdObject={getIdObject}
+                  removeFromWish={removeFromWish}
+                  setWishList={setWishList}
```

Comment:
Try creating a props object (that stores all the different props variables), so you only have to pass one or a couple object to child component